crates/napi/src/next_api/project.rs

```diff
@@ -1099,15 +1099,30 @@ pub async fn project_trace_source(
                 }
             };
 
-            let Some(source_file) = original_file.strip_prefix(SOURCE_MAP_PREFIX) else {
-                bail!("Original file ({}) outside project", original_file)
-            };
+            let project_path_fs = Vc::try_resolve_downcast_type::<DiskFileSystem>(
+                project.container.project().project_path().fs(),
+            )
+            .await?
+            .context("expected project path to be a DiskFileSystem")?;
+
+            let project_path_uri = format!(
+                "file://{}/",
+                sys_to_unix(
+                    &project_path_fs
+                        .await?
+                        .to_sys_path(project.container.project().project_path())
+                        .await?
+                        .to_string_lossy()
+                )
+            );
```

Comment:
A good follow-up would be to double-check this does the right thing on Windows. There could be some unaccounted for edge-case behavior (e.g. is `sys_to_unix` tested with absolute paths?)